### PR TITLE
Add utilities for amp-story component

### PR DESCRIPTION
**Pull request by newmuis**
Adds utility methods and constants for `amp-story` component.

/cc @alanorozco

### CODE REVIEW

File: extensions/amp-story/0.1/events.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+};
+
+
+/**
+ * @param {!Element} source
+ * @param {string} eventName
+ * @param {boolean=} opt_bubbles
+ */
+export function dispatch(source, eventName, opt_bubbles) {
+  const event = new Event(eventName, {bubbles: !!opt_bubbles});
```
Comment (choumx) [inline extensions/amp-story/0.1/events.js]: Use `createCustomEvent` here too. Event constructor doesn't work on IE.

File: extensions/amp-story/0.1/events.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  SWITCH_PAGE: 'ampstory:switchpage',
+};
+
+
+/**
+ * @param {!Element} source
+ * @param {string} eventName
+ * @param {boolean=} opt_bubbles
+ */
+export function dispatch(source, eventName, opt_bubbles) {
```
Comment (choumx) [inline extensions/amp-story/0.1/events.js]: `custom-element.js#dispatchCustomEvent` may obviate this.

File: extensions/amp-story/0.1/events.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+
+/**
+ * @param {!Window} win
+ * @param {!Element} source
+ * @param {string} eventName
+ * @param {!CustomEventInit} eventInit
+ */
+export function dispatchCustom(win, source, eventName, payload, opt_eventInit) {
```
Comment (choumx) [inline extensions/amp-story/0.1/events.js]: Nit: You can combine this with `dispatch()` above.

File: extensions/amp-story/0.1/fullscreen.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+  if (el.requestFullscreen) {
+    el.requestFullscreen();
+  } else if (el.webkitRequestFullScreen) {
+    el.webkitRequestFullScreen();
+  } else if (el.mozRequestFullScreen) {
+    el.mozRequestFullScreen();
+  } else if (el.msRequestFullscreen) {
+    el.msRequestFullscreen();
+  } else {
+    dev().warn('Ignored fullscreen request.');
```
Comment (choumx) [inline extensions/amp-story/0.1/fullscreen.js]: The first arg should be the tag, e.g. `'AMP-STORY'`. Ditto below.

File: extensions/amp-story/0.1/fullscreen.js [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+ * limitations under the License.
+ */
+import {dev} from '../../../src/log';
+
+
+/**
+ * @param {!Element} el
+ */
+const requestFullScreenImpl = function(el) {
+  if (el.requestFullscreen) {
```
Comment (choumx) [inline extensions/amp-story/0.1/fullscreen.js]: Nit: `typeof el.requestFullscreen === 'function'` is a bit more precise.

File: extensions/amp-story/0.1/fullscreen.js [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+      el.mozRequestFullScreen ||
+      el.msRequestFullscreen);
+};
+
+
+/**
+ * @param {!Document} doc
+ * @return {?Element}
+ */
+const getFullscreenElementImpl = function(doc) {
```
Comment (choumx) [inline extensions/amp-story/0.1/fullscreen.js]: Why not use this in `requestFullScreenImpl` above?

File: extensions/amp-story/0.1/fullscreen.js [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+ * @return {boolean}
+ */
+export let isFullScreenSupported = isFullScreenSupportedImpl;
+
+
+/**
+ * @param {!Document} doc
+ * @return {?Element}
+ */
+export const getFullScreenElement = getFullscreenElementImpl;
```
Comment (choumx) [inline extensions/amp-story/0.1/fullscreen.js]: Any reason why we don't directly export the functions above?

```js
export function getFullScreenElement() {
  // ...
}
```

File: extensions/amp-story/0.1/test/test-fullscreen.js [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+  isFullScreenSupported,
+  requestFullScreen,
+} from '../fullscreen';
+
+
+describes.fakeWin('amp-story fullscreen', {
+}, () => {
+  afterEach(() => {
+    sandbox.restore();
+  });
```
Comment (choumx) [inline extensions/amp-story/0.1/test/test-fullscreen.js]: `describes.foo` returns a self-restoring sandbox in the `env` object.

```js
describes.fakeWin('amp-story fullscreen', {}, env => {
  it('...') {
    env.sandbox.stub(...);
  }
});
```

File: extensions/amp-story/0.1/events.js (hunk 3 above)
Comment (newmuis) [reply extensions/amp-story/0.1/events.js]: How about:

1. Delete `dispatch()` above
2. Rename `dispatchCustom()` to `dispatch()`
3. Make `payload` argument optional
4. Update all callers

If that works, would it be okay to file a bug for this and do the refactor as a follow up?

File: extensions/amp-story/0.1/events.js (hunk 1 above)
Comment (newmuis) [reply extensions/amp-story/0.1/events.js]: (see comment below)

File: extensions/amp-story/0.1/events.js (hunk 3 above)
Comment (choumx) [reply extensions/amp-story/0.1/events.js]: SGTM

Comment (newmuis) [reply extensions/amp-story/0.1/events.js]: Filed #11520

**Pull request merged**